spyder_notebook/widgets/client.py

```diff
@@ -123,6 +123,7 @@ def __init__(self, plugin, name):
 
         self.plugin_actions = plugin.get_plugin_actions()
         self.notebookwidget = NotebookWidget(self)
+        self.notebookwidget.selectionChanged.connect(self.hide_header)
```

Comment:
`selectionChanged` is triggered when a click is done in the notebook or when the focus changes (for example when selecting a cell). In this case I used this signal since when you create a new notebook, it sets the focus to the first cell automatically and therefore the signal is triggered.